packages/eddsa-poseidon/src/eddsa-poseidon.ts

```diff
@@ -174,6 +174,60 @@ export function unpackPublicKey(publicKey: BigNumberish): Point<string> {
     return [unpackedPublicKey[0].toString(), unpackedPublicKey[1].toString()]
 }
 
+/**
+ * Packs an EdDSA signature into a buffer of 64 bytes for efficient storage.
+ * Use {@link unpackSignature} to reverse the process without needing to know
+ * the details of the format.
+ *
+ * The buffer contains the R8 point packed int 32 bytes (via
+ * {@link packSignature}) followed by the S scalar.  All encodings are
+ * little-endian.
+ *
+ * @param signature the signature to pack
+ * @returns a 64 byte buffer containing the packed signature
+ */
+export function packSignature(signature: Signature): Buffer {
+    if (!isSignature(signature) || !inCurve(signature.R8) || BigInt(signature.S) >= subOrder) {
+        throw new Error("Invalid signature")
+    }
+
+    const numericSignature: Signature<bigint> = {
+        R8: signature.R8.map((c) => BigInt(c)) as Point<bigint>,
+        S: BigInt(signature.S)
+    }
+
+    const packedR8 = packPoint(numericSignature.R8)
+    const packedBytes = Buffer.alloc(64)
+    packedBytes.set(leBigIntToBuffer(packedR8), 0)
```

Comment:
Should `leBigIntToBuffer` also take the number of bytes as the second parameter here (i.e. size = 32)?

Reply:
Good call.  I forgot that argument was added.

Reply:
Interesting.  In attempting to write a unit test to catch a length issue here, I discovered that the code is actually correct without an explicit length.  Because it's using little-endian byte order, it's okay to have a smaller-than-32-byte buffer set into the buffer, since the rest will already be padded with zeroes at the correct end.  If we were using BE the truncation would ended up with the bytes at the wrong end.  Meanwhile numbers larger than 32 bytes already rules out by the inCurve and > subOrder checks above.

I'll add the explicit size anyway for clarity.

Reply:
Ahh good to know! Thanks for looking into it 👍🏽